### PR TITLE
[Dependencies] Fixes package vulnerabilities

### LINKED ISSUE #1655: Upgrade Cypress
Cypress 
- v10
- v11?

### LINKED ISSUE #1669: Upgrade Dependencies
- `yarn audit`
- add resolutions for any which can't be automatically `yarn upgrade`d

### LINKED ISSUE #1672: Fix Cypress Logout issue - DEV
Certain tests are failing when doing tests strictly off `localhost`. Cypress is unable to `Logout` between tests which is causing an issue where Cypress is automatically getting logged back in and this is causing Cypress to be stuck looking for a `Login` form.

Tests files to fix:
- Filing
- Keycloak
  - Failed login attempt provides valid password reset links
  - Performs form validation
- HMDA Help Institution
  - Can delete and create Institutions

### PR DESCRIPTION
**Pull request by Michaeldremy**
Closes #1669
Closes #1655 
Closes #1672

### CODE REVIEW

File: cypress.config.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    // Excludes file that allows us to run all tests in the UI
+    // Ensures PROD doesn't run all tests twice
+    excludeSpecPattern: !process.env.CYPRESS_ENVIRONMENT
+      ? ['cypress/e2e/all/index.cy.js']
+      : [],
+    setupNodeEvents(on, config) {
+      return require('./cypress/plugins/index.js')(on, config)
+    },
+    specPattern: 'cypress/e2e/**/*.{js,jsx,ts,tsx}',
+  },
```
Comment (meissadia) [inline cypress.config.js:32]: Does the upgraded Cypress version still support the `--spec <list_of_tests>` option? Or does this `specPattern` determine which tests files are executed?
 
We currently use an environment variable with `--spec` to decide which tests to run so I'm just trying to understand if that becomes obsolete or simply updated to reflect the new directory structure.

File: package.json [hunk 2]
```diff
@@ -71,9 +71,9 @@
   "devDependencies": {
     "@cypress/skip-test": "2.6.0",
     "@testing-library/cypress": "8.0.1",
-    "cypress": "9.5.1",
+    "cypress": "11",
```
Comment (meissadia) [inline package.json:74]: For consistency let's add the full semver.  From looking in the `node_modules` folder, it looks like we're already on `11.2.0`.  

```suggestion
    "cypress": "11.2.0",
```

**Review by meissadia — CHANGES_REQUESTED** [review]

File: cypress.config.js (hunk 1 above)
Comment (Michaeldremy) [reply cypress.config.js:32]: Looks like Cypress v11.2.0 re-added the ability to run all spec files when doing `cypress open` aka via the browser. Going to remove this code and the `e2e/all/` directory as it's no longer needed.

File: package.json (hunk 2 above)
Comment (Michaeldremy) [reply package.json]: Has been updated.

**Comment by Michaeldremy** [comment]
Cypress has been upgraded to `12.0.2`. Reason for doing this was to get us to the most up to date version and more importantly fixes #1672.

**Review by meissadia — APPROVED** [review]
We'll merge once we're ready to deploy

**Pull request merged**